### PR TITLE
Added GET /api/drivers/all and DriverController + Tests

### LINKED ISSUE #55: Add backend DriversController with route /api/drivers/all and tests
# Depends on

At least the first two stories from Epic #62 (any user can add their cell phone) must be completed first; i.e the ones that add cell phone to the entity, and add a backend route for updating the cell phone.

# User Story

* As a rider, admin, or driver
* I can see a list of all drivers, along with their cell phone numbers
* So that I can contact drivers as needed

# Acceptance Criteria

- [ ] Add backend DriversController with a route `/api/drivers/all` that gets a list of all users that are drivers.

### PR DESCRIPTION
**Pull request by Aasish-Virjala**
In this PR, I added a GET route /api/drivers/all to get a list of all Users that are Drivers. Only logged in Riders, Drivers, and Admins can access this GET route. Screenshot of swagger below.

Screenshot 2023-08-25 at 4 45 31 AM

Closes #55

### CODE REVIEW

**Comment by Aasish-Virjala** [comment]
Idk why it's showing that 6 files have been changed, even though only 2 have been (DriverController and DriverControllerTests). The other 4 files were from a previous PR but still committed to main, hence why it is showing it, I presume.

**Review by pabucsb — DISMISSED** [review]
LGTM

**Review by pabucsb — DISMISSED** [review]
LGTM

**Review by pabucsb — APPROVED** [review]
LGTM

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriversController.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        for (User user : users) {
+            if(user.getDriver()) {
+                driverUsers.add(user);
+            }
+        }
+        
+        String body = mapper.writeValueAsString(driverUsers);
+        return ResponseEntity.ok().body(body);
+    }
+}
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriversController.java:55]: This is way more complicated than it needs to be.  

```suggestion
    public Iterable<User> drivers()
            throws JsonProcessingException {
        Iterable<User> drivers = userRepository.findAllByDriver(true);
        return drivers;
    }
}
```

**Review by pconrad — CHANGES_REQUESTED** [review]
The code can be much simpler

**Pull request merged**